File: Leetcode/Daily/06-2025/JavaScript/16.js
// Problem: 2016. Maximum Difference Between Increasing Elements (Easy)
// Link: https://leetcode.com/problems/maximum-difference-between-increasing-elements

/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumDifference = function(nums) {
    let n = nums.length, maxx = 0, ans = -1;

    for (let i = n-1; i >= 0; i--) {
        if (maxx > nums[i]) ans = Math.max(ans, maxx - nums[i]);
        maxx = Math.max(maxx, nums[i]);
    }

    return ans;
};

File: Leetcode/Daily/11-2025/C++/16.cpp
// Problem: 1513. Number of Substrings With Only 1s (Medium)
// Link: https://leetcode.com/problems/number-of-substrings-with-only-1s

class Solution {
public:
    int numSub(string s) {
        int n = s.size(), ans = 0, MOD = 1e9 + 7, one = 0;

        for (int i = 0; i < n; i++) {
            if (s[i] - '0') one++;
            else {
                ans = (ans + 1ll * one * (one + 1) / 2) % MOD;
                one = 0;
            }
        }

        ans = (ans + 1ll * one * (one + 1) / 2) % MOD;
        return ans;
    }
};